pkgs/development/libraries/rocprofiler/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, writeScript
+, cmake
+, roctracer
+, hsa-amd-aqlprofile
+, hip
+, numactl
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "rocprofiler";
+  rocmVersion = "5.3.3";
+  version = finalAttrs.rocmVersion;
+
+  src = fetchFromGitHub {
+    owner = "ROCm-Developer-Tools";
+    repo = "rocprofiler";
+    rev = "rocm-${finalAttrs.rocmVersion}";
+    hash = "sha256-e7d78nQYTlMAq6nTcGRhNUGu3Ms1qvv17ET5RTgqKyU=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    hip
+  ];
+
+  buildInputs = [
+    hsa-amd-aqlprofile
+    numactl
+  ];
+
+  cmakeFlags = [
+    "-DCMAKE_C_COMPILER=hipcc"
+    "-DCMAKE_CXX_COMPILER=hipcc"
+    "-DPROF_API_HEADER_PATH=${roctracer.src}/inc/ext"
+  ];
+
+  postPatch = ''
+    patchShebangs test/run.sh
+    patchShebangs bin/*.sh
+  '';
+
+  postInstall = ''
+    patchelf --set-rpath ${lib.makeLibraryPath (finalAttrs.nativeBuildInputs ++ finalAttrs.buildInputs)}:$out/lib $out/lib/rocprofiler/librocprof-tool.so*
```

Comment:
We cannot combine nativeBuildInputs with buildInputs because they have different arches when cross compiling.